HARK/core.py

```diff
@@ -378,31 +503,37 @@ def unpack(self, parameter):
             self.__dict__[parameter].append(solution_t.__dict__[parameter])
         self.add_to_time_vary(parameter)
 
-    def solve(self, verbose=False):
+    # At present, the only solution method is EGM
+    # The next method to add is value function iteration
+    def solve(self, messaging_level=logging.DEBUG, quietly=False, **kwds):  # AgentType
         """
-        Solve the model for this instance of an agent type by backward induction.
-        Loops through the sequence of one period problems, passing the solution
-        from period t+1 to the problem for period t.
+        Solve the model for this instance of an agent by backward induction.
+        Loops through sequence of one period/stage problems, passing solution
+        from next period/stage to the problem for current period/stage.
 
         Parameters
         ----------
-        verbose : boolean
-            If True, solution progress is printed to screen.
+        messaging_level : logging level
+            From logging.DEBUG to logging.CRITICAL
+
+        quietly : boolean
+            If True, suppress output
```

Comment:
There are some circs in which you may want things to run quietly, but not disturb the messaging level set by the user.  It would be a pain (and error-prone) to set `self.messaging_level_originally_requested_by_user` then set the logging level to suppress all output, then to restore `self.messaging_level_originally_requested_by_user.`  There are a couple of other use cases as well, like you might want to suppress output while debugging because you know what it will be and it just clutters things up.